[BUG] Nudge hours params casting error

**Describe the bug**
A clear and concise description of what the bug is.

Currently an error occurs when attempting to read the "hours" param from a nudge. 

```
panic: interface conversion: interface {} is float64, not int32
2022-08-04 17:45:01
goroutine 135 [running]:
2022-08-04 17:45:01
lms/core.nudgesLogic.processLastLoginNudgePerUser(0xc0002f8ac0, 0xe78538, 0xc0001d7c80, 0xe63e00, 0xc0019a7300, 0xe63e20, 0xc0019a7320, 0xe79a88, 0xc0000b0660, 0x0, ...)
2022-08-04 17:45:01
/lms-app/core/app_nudges.go:243 +0x7cc
...
```

**To Reproduce**
Steps to reproduce the behavior:
1. Trigger nudge processing
2. Receive exception

**Expected behavior**
A clear and concise description of what you expected to happen.
When JSON numbers are decoded into an `interface{}` they are parsed as the `float64` type. We need to update the cast type to float to avoid this issue.

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -5,6 +5,8 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
 and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
 
 ## [Unreleased]
+### Fixed
+- Nudge hours params casting error [#35](https://github.com/rokwire/lms-building-block/issues/35)
 
 ## [1.1.9] - 2022-08-04
 ### Added

--- a/core/app_nudges.go
+++ b/core/app_nudges.go
@@ -240,7 +240,7 @@ func (n nudgesLogic) processLastLoginNudgePerUser(nudge model.Nudge, user Groups
 	}
 
 	//determine if needs to send notification
-	hours := float64(nudge.Params["hours"].(int32))
+	hours := nudge.Params["hours"].(float64)
 	now := time.Now()
 	difference := now.Sub(*lastLogin) //difference between now and the last login
 	differenceInHours := difference.Hours()
@@ -333,7 +333,7 @@ func (n nudgesLogic) processMissedAssignmentNudgePerUser(nudge model.Nudge, user
 	}
 
 	//determine for which of the assignments we need to send notifications
-	hours := float64(nudge.Params["hours"].(int32))
+	hours := nudge.Params["hours"].(float64)
 	now := time.Now()
 	missedAssignments, err = n.findMissedAssignments(hours, now, missedAssignments)
 	if err != nil {
@@ -454,7 +454,7 @@ func (n nudgesLogic) processCompletedAssignmentEarlyNudgePerUser(nudge model.Nud
 	}
 
 	//determine for which of the submissions we need to send notifications
-	hours := float64(nudge.Params["hours"].(int32))
+	hours := nudge.Params["hours"].(float64)
 	now := time.Now()
 	ecAssignments, err = n.findCompletedEarlyAssignments(hours, now, ecAssignments)
 	if err != nil {

--- a/driven/storage/collection.go
+++ b/driven/storage/collection.go
@@ -17,9 +17,11 @@ package storage
 import (
 	"context"
 	"errors"
+	"fmt"
 	"log"
 	"time"
 
+	"github.com/rokwire/logging-library-go/logs"
 	"go.mongodb.org/mongo-driver/bson"
 	"go.mongodb.org/mongo-driver/mongo"
 	"go.mongodb.org/mongo-driver/mongo/options"
@@ -203,37 +205,51 @@ func (collWrapper *collectionWrapper) CountDocuments(filter interface{}) (int64,
 	return count, nil
 }
 
-func (collWrapper *collectionWrapper) Watch(pipeline interface{}) error {
+func (collWrapper *collectionWrapper) Watch(pipeline interface{}, l *logs.Logger) {
+	var rt bson.Raw
+	var err error
+	for {
+		rt, err = collWrapper.watch(pipeline, rt, l)
+		if err != nil {
+			l.Errorf("mongo watch error: %s\n", err.Error())
+		}
+	}
+}
+
+//Helper function for Watch
+func (collWrapper *collectionWrapper) watch(pipeline interface{}, resumeToken bson.Raw, l *logs.Logger) (bson.Raw, error) {
 	if pipeline == nil {
 		pipeline = []bson.M{}
 	}
 
-	var opts *options.ChangeStreamOptions
-	opts = options.ChangeStream()
+	opts := options.ChangeStream()
 	opts.SetFullDocument(options.UpdateLookup)
+	if resumeToken != nil {
+		opts.SetResumeAfter(resumeToken)
+	}
 
 	ctx := context.Background()
 	cur, err := collWrapper.coll.Watch(ctx, pipeline, opts)
 	if err != nil {
-		log.Printf("error watching: %s\n", err)
-		return err
+		time.Sleep(time.Second * 3)
+		return nil, fmt.Errorf("error watching: %s", err)
 	}
 	defer cur.Close(ctx)
 
 	var changeDoc map[string]interface{}
-	log.Println("waiting for changes")
+	l.Infof("%s: waiting for changes\n", collWrapper.coll.Name())
 	for cur.Next(ctx) {
 		if e := cur.Decode(&changeDoc); e != nil {
-			log.Printf("error decoding: %s\n", e)
+			l.Errorf("error decoding: %s\n", e)
 		}
 		collWrapper.database.onDataChanged(changeDoc)
 	}
 
 	if err := cur.Err(); err != nil {
-		log.Printf("error cur.Err(): %s\n", err)
-		return err
+		return cur.ResumeToken(), fmt.Errorf("error cur.Err(): %s", err)
 	}
-	return nil
+
+	return cur.ResumeToken(), errors.New("unknown error occurred")
 }
 
 func (collWrapper *collectionWrapper) ListIndexes() ([]bson.M, error) {

--- a/driven/storage/database.go
+++ b/driven/storage/database.go
@@ -98,7 +98,7 @@ func (m *database) start() error {
 	m.nudges = nudges
 	m.sentNudges = sentNudges
 
-	go m.configs.Watch(nil)
+	go m.configs.Watch(nil, m.logger)
 
 	return nil
 }

